Twisting Nether

zumbak04 opened the pull request:
<!--
A basic changelog, goes to patch notes of the next version, so should be as short and informative as possible.
-->
## Changelog:
- Characters with Demon Trait don't die, they return to the Twisting Nether and may be reborn later.
  - Added a background for the Twisting Nether.
  - Being in the Nether makes you isolated, you can't interact much with the world, can't be a target of schemes.
- Added the Incapable game concept.
- The Incapable and In Twisting Nether traits mention they make characters incapable.
- Satyrs look like man'ari for now.

<!--
A changelog where you can describe more complicated things to other developers and testers.
-->
## Developer changelog:
- Replace vanilla death effect with mod override death effect.
- Added `is_incapable_trigger` to replace vanilla trait check,

# Tasks:

<details>
<summary>Click to expand</summary>

- [x] Add Return to Nether effect
- [x] Create Return to Twisting Nether concept. Mentions when you can't return.
- [x] Add Killed Me opinion
- [x] Localize the event
- [x] Add stress, prestige level cost
- [x] Add event and character background
- [x] Relocalize
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/129456303-cb99663f-1428-4dea-ac92-4aabbd4bd2fa.png)

</details>

- [x] This event shouldn't happen
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/129459385-2e524aee-022a-4486-a933-aa0e1d26b9b2.png)

</details>

- [x] You shouldn't be able to plot against them
- [x] Fix
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/129485715-77095711-1758-4a85-b309-a8b377d531dc.png)

</details>

</details>

<!--
If you need to explain something to testers. Otherwise, delete this part.
-->
# How to test:
Start as a demon (like satyr) and try to die.
In the Debug mode, you can right-click your character and force someone to kill you.
Tell your opinion about price and balance.

@Warcraft-GoA-Development-Team/english-proofreaders open the [Page](https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth-2/pull/592/files?file-filters%5B%5D=.yml), scroll to the bottom to the `.yml` files, check them, suggest changes. If you forgot, here are the [Suggestion Guidelines](https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth-2/wiki/How-to-be-Tester-(For-CK3)#reviewing-code).

LordSilvermane commented on localization/english/event_localization/wc_being_events_l_english.yml:
```diff
@@ -6,4 +6,14 @@
+ WCBEI_10_OPT_B:0 "This is not my [ROOT.Char.Custom2('RelationToMeShort', SCOPE.sC('lost_character'))] anymore."
+ 
+ WCBEI_100_TITLE:0 "Twisting Nether"
+ WCBEI_100_DESC:0 "I've lost my body in mortal worlds but my soul found its way to the Twisting Nether.\n\nThousands of twisted demonic souls, merging into a single stream, chase grains of cosmic energy. Only the strongest survive. The rest meet oblivion.\n\nI can join the stream, feed on magic and regain my mortal shell or give up and dive into the Great Dark Beyond."
```
```suggestion
 WCBEI_100_DESC:0 "I've lost my body in the mortal planes but my soul has found its way to the Twisting Nether.\n\nThousands of twisted demonic souls, merging into a single stream, chase grains of cosmic energy. Only the strongest survive. The rest meet oblivion.\n\nI can join the stream, feed on magic and regain my mortal shell or give up and dive into the Great Dark Beyond."
```

TheGroxEmpire replied:
```suggestion
 WCBEI_100_DESC:0 "I've lost my body in mortal worlds but my soul found its way to the Twisting Nether.\n\nThousands of twisted demonic souls, merging into a single stream, chasing grains of cosmic energy. Only the strongest survive. The rest meet oblivion.\n\nI can join the stream, feeding on the magic and regain my mortal shell; otherwise, give up and dive into the Great Dark Beyond."
```

LordSilvermane commented on localization/english/event_localization/wc_being_events_l_english.yml:
```diff
@@ -6,4 +6,14 @@
+ WCBEI_100_TITLE:0 "Twisting Nether"
+ WCBEI_100_DESC:0 "I've lost my body in mortal worlds but my soul found its way to the Twisting Nether.\n\nThousands of twisted demonic souls, merging into a single stream, chase grains of cosmic energy. Only the strongest survive. The rest meet oblivion.\n\nI can join the stream, feed on magic and regain my mortal shell or give up and dive into the Great Dark Beyond."
+ 
+ WCBEI_100_OPT_A:0 "I'll return!"
```
```suggestion
 WCBEI_100_OPT_A:0 "I will return!"
```

LordSilvermane commented on localization/english/event_localization/wc_being_events_l_english.yml:
```diff
@@ -6,4 +6,14 @@
+ WCBEI_100_OPT_A:0 "I'll return!"
+ WCBEI_100_OPT_B:0 "Finally I can rest..."
+ 
+ WCBEI_110_DESC:0 "Time goes differently in the Twisting Nether. I lost track of whether it was decades or centuries.\n\nI have been wandering in the Great Dark Beyond, lurking for magic and my way to Azeroth. Finally. I've found it.\n\nBy force of will, I materialize in mortal worlds in my formal body."
```
```suggestion
 WCBEI_110_DESC:0 "Time goes differently in the Twisting Nether. I've lost track of whether it has been decades or centuries.\n\nI have been wandering in the Great Dark Beyond, lurking for magic and my way back to Azeroth. Finally, I've found it.\n\nBy sheer force of will, I materialize back in the mortal plane in my formal body."
```

LordSilvermane commented on localization/english/wc_game_concepts_l_english.yml:
```diff
@@ -50,3 +50,9 @@
+ game_concept_twisting_nether_desc:0 "Instead of [character_death_dying|E], [characters|E] with [GetTrait('being_demon').GetName( GetNullCharacter )] [trait|E] return to the Twisting Nether.\n\nThey get [GetTrait('in_twisting_nether').GetName( GetNullCharacter )] [trait|E] but can return to Azeroth some time later.\n\nReturning increase [stress|E], decreases [prestige_level|E].\n\nDemons can't re ...
+ 
+ game_concept_wc_incapable:0 "Incapable"
+ game_concept_wc_incapable_desc:0 "Incapable characters cannot take actions independently due to sever illness or being isolated."
```
```suggestion
 game_concept_wc_incapable_desc:0 "Incapable characters cannot take actions independently due to severe illness or being isolated."
```

LordSilvermane reviewed (commented):
Made a few loc suggestions.

TheGroxEmpire commented on localization/english/wc_game_concepts_l_english.yml:
```diff
@@ -50,3 +50,9 @@
  game_concept_necro_government_desc:0 "A [dark_government|E] available to [GetTrait('being_undead').GetName( GetNullCharacter )] with $doctrine_death_magic_name$ accepted. Necro [armies|E] reinforce much faster."
+ 
+ game_concept_twisting_nether:0 "Twisting Nether"
+ game_concept_twisting_nether_desc:0 "Instead of [character_death_dying|E], [characters|E] with [GetTrait('being_demon').GetName( GetNullCharacter )] [trait|E] return to the Twisting Nether.\n\nThey get [GetTrait('in_twisting_nether').GetName( GetNullCharacter )] [trait|E] but can return to Azeroth some time later.\n\nReturning increase [stress|E], decreases [prestige_level|E].\n\nDemons can't re ...
```
```suggestion
 game_concept_twisting_nether_desc:0 "Instead of [character_death_dying|E], [characters|E] with [GetTrait('being_demon').GetName( GetNullCharacter )] [trait|E] returns to the Twisting Nether.\n\nThey get [GetTrait('in_twisting_nether').GetName( GetNullCharacter )] [trait|E] but can return to Azeroth some time later.\n\nReturning increase [stress|E], decreases [prestige_level|E].\n\nDemons can't return if they...\n$EFFECT_LIST_BULLET$Vanish\n$EFFECT_LIST_BULLET$Are sacrificed\n$EFFECT_LIST_BULLET$Suicide\n$EFFECT_LIST_BULLET$Are scared to death"
```
I assume ``[GetTrait('being_demon').GetName( GetNullCharacter )] [trait|E]`` is a third person, singular subject (He, She, It). Thus it should be "returns".

zumbak04 replied:
Nope, it's `Characters with Demon Trait return`.

TheGroxEmpire replied:
Alright, thanks for clarifying

TheGroxEmpire commented on localization/english/event_localization/wc_being_events_l_english.yml:
```diff
@@ -6,4 +6,14 @@
+ WCBEI_10_OPT_B:0 "This is not my [ROOT.Char.Custom2('RelationToMeShort', SCOPE.sC('lost_character'))] anymore."
+ 
+ WCBEI_100_TITLE:0 "Twisting Nether"
+ WCBEI_100_DESC:0 "I've lost my body in the mortal planes but my soul has found its way to the Twisting Nether.\n\nThousands of twisted demonic souls, merging into a single stream, chase grains of cosmic energy. Only the strongest survive. The rest meet oblivion.\n\nI can join the stream, feed on magic and regain my mortal shell or give up and dive into the Great Dark Beyond."
```
```suggestion
 WCBEI_100_DESC:0 "I've lost my body in the mortal planes but my soul has found its way to the Twisting Nether.\n\nThousands of twisted demonic souls, merging into a single stream, chasing grains of cosmic energy. Only the strongest may survive, the rest meet oblivion.\n\nI can join the stream, feeding on magic and regain my mortal shell; otherwise, give up and dive into the Great Dark Beyond."
```

LordSilvermane commented on localization/english/event_localization/wc_being_events_l_english.yml:
```diff
@@ -6,4 +6,14 @@
+ WCBEI_10_OPT_B:0 "This is not my [ROOT.Char.Custom2('RelationToMeShort', SCOPE.sC('lost_character'))] anymore."
+ 
+ WCBEI_100_TITLE:0 "Twisting Nether"
+ WCBEI_100_DESC:0 "I've lost my body in the mortal planes but my soul has found its way to the Twisting Nether.\n\nThousands of twisted demonic souls, merging into a single stream, chasing grains of cosmic energy. Only the strongest may survive, the rest meet oblivion.\n\nI can join the stream, feeding on magic and regain my mortal shell; otherwise, give up and dive into the Great Dark Beyond."
```
I know TheGroxEmpire suggested this one, but I personally think it should be reverted. The semicolon isn't necessary, and the `or` makes it clearer that it is a choice between the two options.
```suggestion
 WCBEI_100_DESC:0 "I've lost my body in the mortal planes but my soul has found its way to the Twisting Nether.\n\nThousands of twisted demonic souls, merging into a single stream, chasing grains of cosmic energy. Only the strongest may survive, the rest meet oblivion.\n\nI can join the stream, feeding on magic and regain my mortal shell or give up and dive into the Great Dark Beyond."
```

LordSilvermane reviewed (commented):
I did want to make one slight revision to TheGroxEmpire's suggestion, but otherwise it looks good.

LordSilvermane reviewed (approved):
All looks good!

StarsPotato reviewed (approved):
I have tested as Ilidan, it works fine and it looks cool!

arithon reviewed (approved):
loc LGTM

Remerod reviewed (changes requested):
So i tested it out.

<details>
<summary>Click to expand</summary>

Are Satyr's meant to look like man'ari in this version?
![image](https://user-images.githubusercontent.com/79544635/130367905-5b7ff237-4c69-41eb-9764-ba7dc8f0266c.png)

So i tried King Kazzak Out, he seam's to have a man'ari beard
![image](https://user-images.githubusercontent.com/79544635/130367959-6c091732-74b4-483e-9417-0511818be9cb.png)

So I used the debug mode to **Slay** him
![image](https://user-images.githubusercontent.com/79544635/130367978-de3ecf8b-77c9-4d7a-b2df-76f6d78f5870.png)

I did not have any renown... so i got killed by a kobold(lost the pic) and now i have to find a new character
![image](https://user-images.githubusercontent.com/79544635/130368004-238b4d1a-1b30-40bd-9750-50db85af2c17.png)

I have found myself a new character with a weird beard
![image](https://user-images.githubusercontent.com/79544635/130368422-eb97ed66-2c96-451c-9565-557ca2ae4607.png)

And adding some more prestige and renown 
![image](https://user-images.githubusercontent.com/79544635/130368437-d5868cc2-8bcb-4c17-9fd8-c51915a2824a.png)

Now I get the choice of return at least.
![image](https://user-images.githubusercontent.com/79544635/130368508-f054ebff-06a8-48d4-93c6-af5b40beaa50.png)

Hanging in the Nether
![image](https://user-images.githubusercontent.com/79544635/130368554-63484ed8-8e2a-4850-aa50-42530d3dcd19.png)

Hmm, can still build and stuff
![image](https://user-images.githubusercontent.com/79544635/130368583-a1d770bf-b0dc-4748-a0c5-8db0dd326bbd.png)

But not declare War
![image](https://user-images.githubusercontent.com/79544635/130368592-f8e12ef5-a035-40e2-985d-4ec0c564ee07.png)

So I waited around for 10 years give and take a couple. And I have to say, i would want some further information for the player on **when** or **how** to come back, because i had no idea, i looked around and couldn't find any information about that. So something about estimating your return. or something that tell you that you will come back, becouse the trait say's that you will be summoned back while the event says you enter back on your own.
![image](https://user-images.githubusercontent.com/79544635/130368900-b47e493d-55fb-4897-83f6-e6119c92d7f1.png)

So overall I think we **need** some better information about how your whereabouts and return from the Twisting Nether. But i would like some more about the realm gameplay while you, the liege is gone. I would like to suggest A Regent/Herald could rule in your place, and event's between you and them could be a thing, like asking them to declare a war, or gather resources for a summoning you back to Azeroth. But with the exception of the lack of your return from the twisting nether i think this mechanic works, and is a favorable gameplay for a demon. I am not much big on balance, but i do like that being cast into the nether or just vanishes is based on renown. but this need also to be strongly pointed out to the player, since this is your lifeblood instead of heir's and player's need to be aware of this, witch is currently not really obvious and to my understanding unavailable before getting the event. Even here on this pic below it is not told;
![image](https://user-images.githubusercontent.com/79544635/130369134-c415e338-6242-481f-bb52-0feb05678910.png)

So to close of my test review, I really like this mechanic, it gives a an unique gamplay for the mod and are fitting for the demons. but i do think it need's a little work to be better playable and understandable for players unfamiliar with the mod.

</details>

ValianBlue reviewed (changes requested):
_**The Twisting Nether Effect**_

_Overall this is an interesting twist on CK's succession system. I find the fame cost, stress gain, and negative effects under the trait itself both make sense and are fair. I would even suggest adding more negative effects such as a negative opinion effect for your vassals and popular opinion, but this is not absolutely necessary to do._

_On my character's first death, they came back after 10 years. My second character came back after 7. In the grand scheme of Crusader Kings 3, this is a pitiful amount of time for an immortal character. However, in the context of this being a WarCraft overhaul (with most of the events of the RTS games and WoW spanning less than a hundred years), this feels impactful._

_I tested to see if the AI could enter into the Twisting Nether on its own after slaying them, and it did successfully on all on my tests._

_The incapable trait is solid for preventing most character actions. Although I had an idea during my playthrough of having an offscreen character (Twisting Nether) that also imprisons you. This is something up for debate, and the incapable trait itself could be sufficient on its own._

_I'd like to add onto what @Remerod suggested. I see the potential for a custom mechanic, sort of like a progress bar that you can influence whilst you're in the Twisting Nether. You can hasten your return by selecting options that give you more progress, but come with penalties. Stress would be a solid one to add, but perhaps piety, having your character be wounded or maimed when they return, all sorts of things like that! Alternatively, the player can choose to go "At their own pace," which means it'll take longer for them to return, but they won't be penalized._

_For a regent, I think it would be interesting if the regent could plot to overthrow you while you're in the Nether. Thus giving you another reason to hasten your return so you keep your titles and claims._

_**Observations**_

_Characters in the Twisting Nether can be captured and imprisoned. An immunity of some sort should be added to the trait to prevent this._

<details>
<summary></summary>

![image](https://user-images.githubusercontent.com/40177753/130370959-2d6fb2dd-348d-4f56-a03f-7cafb2f9dc49.png)
</details>

_Stress can still be gained. It's feasible that a demon soul in the tumultuous Twisting Nether would be vulnerable to stress, but I think this is worth discussing further. A character in the physical world should never be able to affect a demon directly, unless through special circumstances. Actions that affect a demon's realm applying stress would be understandable._

<details>
<summary></summary>

![image](https://user-images.githubusercontent.com/40177753/130371038-cc16e19f-9142-43ed-b99b-b6ca4f60d1bf.png)
</details>

_Diseases can also affect a demon while in the Twisting Nether._

<details>
<summary></summary>

![image](https://user-images.githubusercontent.com/40177753/130371289-af9a7186-148d-4eb6-b942-52d6ba1da609.png)
</details>

I checked to see if I could assassinate a demon in the Twisting Nether, but this is not possible. This is a good thing to see.


<details>
<summary></summary>

![image](https://user-images.githubusercontent.com/40177753/130372263-314bb4c6-0479-449a-9dad-c60ad5dfc6f0.png)
</details>

TheGroxEmpire reviewed (commented):
PS: I deleted the comment so I could replace the resolved convo with a new one on the same loc premise.

Other than this, all else looks good!

arithon commented on localization/english/wc_game_concepts_l_english.yml:
```diff
@@ -50,3 +50,9 @@
  game_concept_necro_government_desc:0 "A [dark_government|E] available to [GetTrait('being_undead').GetName( GetNullCharacter )] with $doctrine_death_magic_name$ accepted. Necro [armies|E] reinforce much faster."
+ 
+ game_concept_twisting_nether:0 "Twisting Nether"
+ game_concept_twisting_nether_desc:0 "Instead of [character_death_dying|E], [characters|E] with [GetTrait('being_demon').GetName( GetNullCharacter )] [trait|E] return to the Twisting Nether.\n\nThey get [GetTrait('in_twisting_nether').GetName( GetNullCharacter )] [trait|E] but can return to Azeroth some time later.\n\nReturning increase [stress|E], decreases [prestige_level|E].\n\nDemons can't re ...
```
If any, "Committed suicide"

zumbak04 commented:
> Are Satyr's meant to look like man'ari in this version?

It's said in the changelog.

> So i tried King Kazzak Out, he seam's to have a man'ari beard
Fixed in 36bb22a61e6c4ea202a898f282de9d7e5791deb5.

Added in d0ebe863b07421567a38423294d5e8a28efbc40e.

> And I have to say, i would want some further information for the player on when or how to come back

Added in d0ebe863b07421567a38423294d5e8a28efbc40e.

> I would like to suggest A Regent/Herald could rule in your place

Leave it to the vanilla.

> but this need also to be strongly pointed out to the player, since this is your lifeblood instead of heir's and player's need to be aware of this

Mentioned in d0ebe863b07421567a38423294d5e8a28efbc40e that you pay 2 levels of fame.

zumbak04 commented:
> I would even suggest adding more negative effects such as a negative opinion effect for your vassals and popular opinion, but this is not absolutely necessary to do.

To make them revolt? Control and dread debuffs do it. Lowering control makes your army weaker. Lowering dread unlocks revolts.

> having an offscreen character (Twisting Nether) that also imprisons you

That's intended. You still can imprison.

> I'd like to add onto what @Remerod suggested. I see the potential for a custom mechanic, sort of like a progress bar that you can influence whilst you're in the Twisting Nether.

You can suggest it in the issues, but I don't have such plans for now.

zumbak04 commented:
> Characters in the Twisting Nether can be captured and imprisoned. An immunity of some sort should be added to the trait to prevent this.

Blocked in 2a06de9a175ba87a047e071d42e26129700e43ae.

> Stress can still be gained. It's feasible that a demon soul in the tumultuous Twisting Nether would be vulnerable to stress, but I think this is worth discussing further. A character in the physical world should never be able to affect a demon directly, unless through special circumstances. Actions that affect a demon's realm applying stress would be understandable.

Blocked in eefdde8c928e056dd417bf9b5848d082ec7a9e76.

> Diseases can also affect a demon while in the Twisting Nether.

It's mentioned here https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth-2/issues/593 and should be fixed on separate PR.

ValianBlue reviewed (approved):
_Changes reviewed, tested, and approved._

_New feature suggested in #613._

TheGroxEmpire reviewed (approved):
Looks good to me!

Remerod reviewed (approved):
_Look's fine to me._

But i want to point this out on a side-note ->
<details>
<summary>Click to expand</summary>

The council seems to lack arms, while you yourself have it in the events, is this a bug or a unfortunate limitation on how eredruin are implemented as static objects?
**No arms**
![image](https://user-images.githubusercontent.com/79544635/130510040-e4f5bf70-0b16-4727-86a5-66e434a32e6a.png)
**Arms**
![image](https://user-images.githubusercontent.com/79544635/130510072-c7203256-0a07-42ad-a8fe-1c04e11e21aa.png)

I dunno if this have anything to do with the twisting nether pull, but i wanted to mention it anyway ;)

</details>
